api/v1alpha1/secretagentconfiguration_types.go

```diff
@@ -123,6 +124,7 @@ type AppConfig struct {
 	SecretsManager          SecretsManager `json:"secretsManager" validate:"required,oneof=none GCP AWS"`
```

Comment:
need to add Azure in the `oneof` list. `"required,oneof=none GCP AWS Azure"`
Also:
1. Add an entry for Azure in SecretAgentConfigurationStatus #L36 (still not populated, but will)
1. Add azure to CRD field validation #L76 `// +kubebuilder:validation:Enum=none;GCP;AWS,Azure`